ci: hclogvet does not run on entire codebase

[hclogvet](https://github.com/hashicorp/go-hclog/tree/main/hclogvet) is currently triggered within CI (via our makefile) using the command `hclogvet .` which does not run across all sub-directories within the Nomad project. In order to run this correctly, hclogvet should be triggered with the argument `./...` and when done, the following errors are found:

```console
client/widmgr/widmgr.go:124:16: invalid number of log arguments to Warn (1 valid pair only)
# github.com/hashicorp/nomad/client/allocrunner/taskrunner
# [github.com/hashicorp/nomad/client/allocrunner/taskrunner]
client/allocrunner/taskrunner/api_hook.go:100:19: invalid number of log arguments to Debug (1 valid pair only)
client/allocrunner/taskrunner/dynamic_users_hook.go:80:17: invalid number of log arguments to Error (1 valid pair only)
client/allocrunner/taskrunner/identity_hook.go:137:19: invalid number of log arguments to Error (1 valid pair only)
client/allocrunner/taskrunner/identity_hook.go:151:19: invalid number of log arguments to Warn (1 valid pair only)
# github.com/hashicorp/nomad/client
# [github.com/hashicorp/nomad/client]
client/alloc_endpoint.go:278:18: invalid number of log arguments to Info (1 valid pair only)
client/client.go:1335:18: invalid number of log arguments to Error (2 valid pairs only)
# github.com/hashicorp/nomad/scheduler
# [github.com/hashicorp/nomad/scheduler]
scheduler/scheduler_system.go:365:18: invalid number of log arguments to Debug (1 valid pair only)
# github.com/hashicorp/nomad/nomad
# [github.com/hashicorp/nomad/nomad]
nomad/job_endpoint.go:2110:17: invalid number of log arguments to Error (1 valid pair only)
nomad/leader.go:2627:15: invalid number of log arguments to Error (1 valid pair only)
nomad/leader.go:2656:15: invalid number of log arguments to Error (1 valid pair only)
nomad/locks.go:60:17: invalid number of log arguments to Error (1 valid pair only)
nomad/locks.go:149:17: invalid number of log arguments to Error (1 valid pair only)
```

All but one error is easily fixable. The problem found at `client/alloc_endpoint.go:278` is a false positive as hclogvet is unable to understand the log line is being passed an array of k/v pairs as its arguments. hclogvet will need to be updated, before CI can be modified to run the command correctly.

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -145,7 +145,7 @@ lint-deps: ## Install linter dependencies
 	@echo "==> Updating linter dependencies..."
 	go install github.com/golangci/golangci-lint/v2/cmd/golangci-lint@v2.3.0
 	go install github.com/client9/misspell/cmd/misspell@v0.3.4
-	go install github.com/hashicorp/go-hclog/hclogvet@feaf6d2ec20fd895e711195c99e3fde93a68afc5
+	go install github.com/hashicorp/go-hclog/hclogvet@bd6194f1f5b126dbad2a3fdf3b9b6556cc3496c3
 
 .PHONY: git-hooks
 git-dir = $(shell git rev-parse --git-dir)
@@ -163,7 +163,7 @@ check: ## Lint the source code
 	@cd ./api && golangci-lint run --config ../.golangci.yml --build-tags "$(GO_TAGS)"
 
 	@echo "==> Linting hclog statements..."
-	@hclogvet .
+	@hclogvet ./...
 
 	@echo "==> Spell checking website..."
 	@misspell -error -source=text website/content/

--- a/client/allocrunner/networking_cni.go
+++ b/client/allocrunner/networking_cni.go
@@ -594,10 +594,10 @@ func (c *cniNetworkConfigurator) Teardown(ctx context.Context, alloc *structs.Al
 		// best effort cleanup ipv6
 		ipt, iptErr := c.newIPTables(structs.NodeNetworkAF_IPv6)
 		if iptErr != nil {
-			c.logger.Debug("failed to detect ip6tables: %v", iptErr)
+			c.logger.Debug("failed to detect ip6tables", "error", iptErr)
 		} else {
 			if err := c.forceCleanup(ipt, alloc.ID); err != nil {
-				c.logger.Warn("ip6tables: %v", err)
+				c.logger.Warn("failed to cleanup iptables", "error", err)
 			}
 		}
 

